emloop/cli/common.py

```diff
@@ -213,7 +213,28 @@ def create_hooks(config: dict, model: AbstractModel,
     return hooks
 
 
-def run(config: dict, output_root: str, restore_from: str=None, eval: Optional[str]=None) -> None:
+def create_main_loop(config: dict, model: Optional[AbstractModel]=None, dataset: Optional[AbstractDataset]=None,
+                     hooks: Optional[List[AbstractHook]]=None, eval_stream: Optional[str]=None):
```

Comment:
Wouldn't it be better to pass `extra_streams` to be consistent with the corresponding config option?

The documentation for this parameter should also say what is the default behavior.